src/bot/create_bot.py

```diff
@@ -48,11 +48,17 @@ def create_bot(post_storage: PostStorage, warden: Warden) -> TelegramClient:
         bot_token=TELEGRAM_BOT_TOKEN
     )
 
-    # TODO: should we parse album here?
+    # TODO: should we parse album here? nea
     @bot.on(events.NewMessage(chats=AGGREGATOR_CHANNEL))
     async def aggregator_channel_listener(event) -> None:
         if hasattr(event, "message"):
-            post_storage.post(event.message)
+            message = event.message
+            if message.fwd_from and message.fwd_from.from_id:
+                original_chat = await event.client.get_entity(message.fwd_from.from_id)
+                channel_name = original_chat.title if original_chat else "Unknown Channel"
+                post_storage.post(event.message, channel_name)
```

Comment:
I've looked at telthon's source code to understand what is `title` and found a beautiful function. A suppose we should use it

```python
def get_display_name(entity):
    """
    Gets the display name for the given :tl:`User`,
    :tl:`Chat` or :tl:`Channel`. Returns an empty string otherwise.
    """
```

Also in my opinion, it is better to keep this logic inside PostStorage.post

Reply:
Changed

Reply:
Bro second part of the message :(